generators/example/src/main/scala/ConfigMixins.scala

```diff
@@ -37,129 +41,63 @@ class WithGPIO extends Config((site, here, up) => {
     GPIOParams(address = 0x10012000, width = 4, includeIOF = false))
 })
 
-// ----------------------------------------
-// Rocket Top Level System Parameter Mixins
-// ----------------------------------------
-
-/**
- * Class to specify a "plain" top level rocket-chip system
- */
-class WithNormalRocketTop extends Config((site, here, up) => {
-  case BuildRocketTop => (clock: Clock, reset: Bool, p: Parameters) => {
-    Module(LazyModule(new RocketTop()(p)).module)
-  }
-})
-
-/**
- * Class to specify a top level rocket-chip system with PWM
- */
-class WithPWMRocketTop extends Config((site, here, up) => {
-  case BuildRocketTop => (clock: Clock, reset: Bool, p: Parameters) =>
-    Module(LazyModule(new RocketTopWithPWMTL()(p)).module)
-})
-
-/**
- * Class to specify a top level rocket-chip system with a PWM AXI4
- */
-class WithPWMAXI4RocketTop extends Config((site, here, up) => {
-  case BuildRocketTop => (clock: Clock, reset: Bool, p: Parameters) =>
-    Module(LazyModule(new RocketTopWithPWMAXI4()(p)).module)
-})
-
-/**
- * Class to specify a top level rocket-chip system with a block device
- */
-class WithBlockDeviceModelRocketTop extends Config((site, here, up) => {
-  case BuildRocketTop => (clock: Clock, reset: Bool, p: Parameters) => {
-    val top = Module(LazyModule(new RocketTopWithBlockDevice()(p)).module)
-    top.connectBlockDeviceModel()
-    top
-  }
-})
-
-/**
- * Class to specify a top level rocket-chip system with a simulator block device
- */
-class WithSimBlockDeviceRocketTop extends Config((site, here, up) => {
-  case BuildRocketTop => (clock: Clock, reset: Bool, p: Parameters) => {
-    val top = Module(LazyModule(new RocketTopWithBlockDevice()(p)).module)
-    top.connectSimBlockDevice(clock, reset)
-    top
-  }
-})
-
-/**
- * Class to specify a top level rocket-chip system with GPIO
- */
-class WithGPIORocketTop extends Config((site, here, up) => {
-  case BuildRocketTop => (clock: Clock, reset: Bool, p: Parameters) => {
-    val top = Module(LazyModule(new RocketTopWithGPIO()(p)).module)
-    for (gpio <- top.gpio) {
-      for (pin <- gpio.pins) {
-        pin.i.ival := false.B
-      }
-    }
-    top
-  }
-})
-
-// --------------------------------------
-// BOOM Top Level System Parameter Mixins
-// --------------------------------------
+// -----------------------------------------------
+// BOOM + Rocket Top Level System Parameter Mixins
```

Comment:
The use of `+` and `and` is confusing if my earlier comment about rocket only configs is not wrong. Should this be Boom And/Or Rocket? Do these configs work fro BoomOnly, RocketOnly, and BoomPlusRocket, or some subset of those?

Reply:
Ok. I can make this clearer. This for BOOM and/or Rocket.

Reply:
The functionality provided by the `BoomAndRocket` named things allows you to do a Rocket only, BOOM only, and Rocket+BOOM system.